lib/ex_doc.ex

```diff
@@ -25,6 +25,7 @@ defmodule ExDoc do
       project: nil,
       retriever: ExDoc.Retriever,
       source_beam: nil,
```

Comment:
> I will be submitting another PR, and because we are using nil I need to check agains nil, false or "" to see if an option is unset.

We should continue only checking for `nil`. In Elixir, `nil` is used for absence of information (except if we have a field that may be boolean).

If the field can be an empty string and an empty string also mean "emptiness", we need to convert the empty string to nil before putting it in the struct.